packages/cashier/src/stores/payment-agent-store.js

```diff
@@ -94,8 +94,10 @@ export default class PaymentAgentStore {
     async getPaymentAgentList() {
         // wait for get_settings so residence gets populated in client-store
         // TODO: set residence in client-store from authorize so it's faster
-        await this.WS.wait('get_settings');
-        const { residence, currency } = this.root_store.client;
+        const {
+            get_settings: { country_code: residence },
+        } = await this.WS.wait('get_settings');
+        const { currency } = this.root_store.client;
         return this.WS.authorized.paymentAgentList(residence, currency);
```

Comment:
Earlier we used the `residence` from client_store and populate it with "get_settings" request. Now the reason why we grab the value from the response is because mobx update the `residence` value to late (and our `paymentAgentList(residence, currency)` request has the `residence` is equal to "empty string". There is a TODO to set residence in client-store from authorize response so it will update **much faster!**. It  solves our issues since the authorize request is the **first request** and there we have `country` value which we can use as a residence. But since this is hotfix and checking the new implementation of residence takes QA too much time to check everything in app, for now we use the solution above.

Reply:
The same thing we do with `currency`, we populate it during `authorize` request, and it always has an updated value.